src/components/Api.js

```diff
@@ -0,0 +1,145 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseURL = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkTheApiResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserData() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => this._checkTheApiResponse(res))
```

Comment:
**COULD BE IMPROVED**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkTheApiResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Do I understand it correct, that in this case I'll need to pass the whole options field into the function call, 
For example, in editProfileData call I'll need to pass it 

```
  editProfileData(userInput) {
    return fetch(`${this._baseURL}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name: userInput.name,
        about: userInput.description,
      }),
    })
      .then((res) => this._checkTheApiResponse(res))
      .catch((err) => console.error(err));
  `}`
  
  ```
  to 
  
```
  editProfileData(userInput) {
  _request(`${this._baseURL}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name: userInput.name,
        about: userInput.description,
      }),
    })
      .then((res) => this._checkTheApiResponse(res))
}

  _request(url, options) {
    return fetch(url, options).then(this._checkTheApiResponse)
  }
```

Reply:
> Do I understand it correct, that in this case I'll need to pass the whole options field into the function call,
> For example, in editProfileData call I'll need to pass it

Right


You don't change anything except for `fetch` and remove duplication of `_checkTheApiResponse`